### PR TITLE
Update to Go 1.22 and update deps

### PR DESCRIPTION
**Pull request by marco-m-pix4d**

### CODE REVIEW

File: sets/sets.go [hunk 1]
```diff
@@ -79,10 +79,3 @@ func (s *Set[T]) Difference(x *Set[T]) *Set[T] {
 	}
 	return result
 }
-
-func max(a, b int) int {
```
Comment (aliculPix4D) [inline sets/sets.go:83]: :)

**Review by aliculPix4D — APPROVED** [review]

File: sets/sets.go (hunk 1 above)
Comment (marco-m-pix4d) [inline sets/sets.go:83]: This "simple" change needed a lot of work on the Go compiler...

Comment (SpectreVert) [reply sets/sets.go:83]: Why is that? I looked for articles about it but didn't find much.

**Review by SpectreVert — APPROVED** [review]

**Review by iAmoric — APPROVED** [review]

File: sets/sets.go (hunk 1 above)
Comment (marco-m-pix4d) [reply sets/sets.go:83]: https://go.dev/ref/spec#Min_and_max

```
> The built-in functions min and max compute the smallest—or largest,
> respectively—value of a fixed number of arguments of ordered types
                                                       ^^^^^^^^^^^^^
```
https://go.dev/ref/spec#Comparison_operators defines "ordered types".

So the signature of the built-in max() is 
```
func max[T cmp.Ordered](x T, y ...T) T
```

and... `T` is a "type parameter", that is, Go "generics" ...

**Review by odormond — APPROVED** [review]

**Pull request merged**